dev-tools/mage/integtest_docker.go

```
@@ -68,14 +68,11 @@ func (d *DockerIntegrationTester) Use(dir string) (bool, error) {
 	return false, nil
 }
 
-// HasRequirements ensures that the required docker and docker-compose are installed.
+// HasRequirements ensures that the required docker is installed.
 func (d *DockerIntegrationTester) HasRequirements() error {
 	if err := HaveDocker(); err != nil {
 		return err
 	}
-	if err := HaveDockerCompose(); err != nil {
```

Wonder if `docker compose` could be also validated, it might end up that `docker` cli is installed but people still uses `docker-compose`? I know the CI Images won't have that version, but maybe our dev teams do.

---

> I'd say we need to be aligned here. If the docker-compose is deprecated, it shouldn't be used to avoid any occasions aka "it works on my machine, but fails on the CI".

Indeed

> The removed HaveDockerCompose function was checking for the directory docker-compose, which is not the case for the current version of docker cli. It will require to hardcode the dir path, since it's not in path, and I'm not sure it's worth checking if the compose argument works.

I'd expect `HaveDockerCompose` to validate if `docker compose` is a valid call instead, if `HaveDockerCompose` fail fast it can save people's time in my opinion. Or any misconfiguration in the CI VM Images

---

There's a [test](https://github.com/elastic/beats/blob/33e468bed248a6507c208243969b517e4727a2bd/dev-tools/mage/common.go#L188-L217) that calls `docker info -f`:
```
// HaveDocker returns an error if docker is unavailable.
func HaveDocker() error {
	if _, err := GetDockerInfo(); err != nil {
		return fmt.Errorf("docker is not available: %w", err)
	}
	return nil
}

```
If there's a problem with the docker cli, it will fail. Since both `info` and `compose` are arguments now, I still think that there's no need to check it separately.

---

There is a possiblity `docker compose` not available while `docker version` yes, but I guess, we don't need to do any extra validation as the tests will fail anyway and docker-compose is not a valid binary. IIUC, `docker-cli choco package` installs the docker compose plugin, and it's by default installed when using the desktop:
- https://docs.docker.com/compose/install/

Let's move forward anyway for now